### DISCUSSION TITLE
server not run localhost:3000

### DISCUSSION THREAD

**Discussion by Ahmedsamir190** [vercel/next.js#48782] [Help]
### Summary

Good evening

I am a new user of next.js and I followed the steps to create a new project
 1-npx create-next-app name here After completion,
 I open the project using visual studio code Open the terminal and type npm run dev and it appears
the ready - started server on 0.0.0.0:3000, URL: http://localhost:3000
But I don't see compiled client and server successfully 
the automatic file named .next containing package.json that contains {"type": "commonjs"} only is created 
when pressed to follow 3000 appears 
This site can’t be reached localhost refused to connect.

I have seen that the file (.next) that automatically walks is missing for me, but I don't know why
1_ I turn off the Windows firewall and antivirus
2- I changed the port
3- I deleted everything once and started from the beginning, Of course, I don't owe any error in the console

next: 13.3.1
node:18.16

**Comment by icyJoseph** [comment]
There's some kind of issue where people have installed Node.js version that's incompatible with the OS, regardless of the actual system architecture. Please run `node -p 'process.arch'`, on your terminal, that should output the Windows architecture, x64 or x86, etc. If not, reinstall Node.js, using the correct version this time.

**Reply by Ahmedsamir190** [reply]
I used that command and already uninstalled Node and installed it again according to Windows architecture Knowing that I use React without any problems, but with next, it does not work

**Reply by icyJoseph** [reply]
Alright, and going to `localhost:3000` does nothing either, right? The compiler is lazy, so it won't do any work until the page is required.

Have you done any changes to the initial project setup? 

Could you run `npx next info` and share the results? (Delete anything you might consider sensitive)

**Reply by Ahmedsamir190** [reply]
Correct when going to localhost:3000 I saw this message (This site can’t be reached localhost refused to connect.) I did not change anything in the  initial project
That results for npx next info

    Operating System:
      Platform: win32
      Arch: ia32
      Version: Windows 7 Ultimate Service Pack 1
    Binaries:
      Node: 18.16.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.3.2-canary.6
      eslint-config-next: 13.3.1
      react: 18.2.0
      react-dom: 18.2.0

**Reply by icyJoseph** [reply]
What command are you issuing to run the app? 

I find it strange that neither, `npm`, `yarn`, nor `pnpm`, yield anything 😮

**Reply by Ahmedsamir190** [reply]
what do you mean? I used npm run dev to run the app  
What's wrong and what should I do?

**Reply by Ahmedsamir190** [reply]
PS D:\Work\next\my-app> npm run dev

> my-app@0.1.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
PS D:\Work\next\my-app> node -v
v18.16.0
PS D:\Work\next\my-app> npm -v
9.5.1
PS D:\Work\next\my-app> npx next info

    Operating System:
      Platform: win32
      Arch: ia32
      Version: Windows 7 Ultimate Service Pack 1
    Binaries:
      Node: 18.16.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.3.2-canary.6
      eslint-config-next: 13.3.1
      react: 18.2.0
      react-dom: 18.2.0

**Reply by icyJoseph** [reply]
When I do `next info` I see this in binaries:

```console
    Binaries:
      Node: 18.14.1
      npm: 9.3.1
      Yarn: 1.22.19
      pnpm: 7.22.0
```

I wonder if it is just a false flag, but could be the reason?

Does `npm run build` do something at all? Maybe some other logs?

**Reply by Ahmedsamir190** [reply]
when I do npm run build 

my-app> npm run build

> my-app@0.1.0 build
> next build

info  - Linting and checking validity of types
info  - Downloading WASM swc package...
info  - Using wasm build of next-swc
info  - Creating an optimized production build ..PS D:\Work\next\my-app>

**Reply by icyJoseph** [reply]
Could you add a .babelrc file? With these contents:

```
{
  "presets": ["next/babel"],
  "plugins": []
}
```

I wonder if there's something going on with swc.

**Reply by Ahmedsamir190** [reply]
I create a babelrc file and put this in the file 
{
  "presets": ["next/babel"],
  "plugins": []
}

and run npm run dev and result 
PS D:\Work\next\my-app> npm run dev

> my-app@0.1.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Disabled SWC as replacement for Babel because of custom Babel configuration ".babelrc" https://nextjs.org/docs/messages/swc-disabled
info  - Using external babel configuration from D:\Work\next\my-app\.babelrc
event - compiled client and server successfully in 43.1s (199 modules)
and go to localhost:3000 but didn't work

**Reply by Ahmedsamir190** [reply]
and  when I run npm run dev generates a .next file with have server and static folder and some of the files also 
but still, when I go to localhost:3000 show This site can’t be reached localhost refused to connect.

**Reply by icyJoseph** [reply]
I wonder if there's some other issue then. Try to go to 127.0.0.1:3000

**Reply by Ahmedsamir190** [reply]
I didn't show any issue, for now, no error in the console or terminal just the localhost still didn't work
I try to go 127.0.0.1:3000 don't work also

**Reply by Ahmedsamir190** [reply]
just when i run npm run div see that 
PS D:\Work\next\my-app> npm run dev

> my-app@0.1.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Disabled SWC as replacement for Babel because of custom Babel configuration ".babelrc" https://nextjs.org/docs/messages/swc-disabled
info  - Using external babel configuration from D:\Work\next\my-app\.babelrc
event - compiled client and server successfully in 48.6s (199 modules)
PS D:\Work\next\my-app>

**Reply by Ahmedsamir190** [reply]
now when I run npm run build  

result PS D:\Work\next\my-app> npm run  build

> my-app@0.1.0 build
> next build

info  - Linting and checking validity of types
info  - Using wasm build of next-swc
info  - Disabled SWC as replacement for Babel because of custom Babel configuration ".babelrc" https://nextjs.org/docs/messages/swc-disabled
info  - Using external babel configuration from D:\Work\next\my-app\.babelrc
Failed to compile.

./src/pages/index.js:3:1
Syntax error: "next/font" requires SWC although Babel is being used due to a custom babel config being present.
Read more: https://nextjs.org/docs/messages/babel-font-loader-conflict

info  - Creating an optimized production build .PS D:\Work\next\my-app>

**Reply by icyJoseph** [reply]
A wild idea, but if you changed your `Work/next` directory to `Work/next-projects` (with Windows backslashes), would that work?

**Reply by Ahmedsamir190** [reply]
unfortunately, I changed the directory but still the problem

**Reply by icyJoseph** [reply]
Would you mind posting a screenshot of running the dev command? 

I wonder if there's lack of writing permissions to write to the file system, or to open an http socket. Really strange.

**Reply by Ahmedsamir190** [reply]
![one](https://user-images.githubusercontent.com/109394049/234396215-7ff21a1b-fdc5-4690-a722-48608d628b8b.PNG)
![two](https://user-images.githubusercontent.com/109394049/234396295-1029c047-b249-4f6d-abeb-f979634f926c.PNG)
![three](https://user-images.githubusercontent.com/109394049/234396359-41dbd290-a378-4060-aa2a-60abcd61cc5f.PNG)

**Reply by Ahmedsamir190** [reply]
if you went screenshot to specific thing tell me

**Reply by Ahmedsamir190** [reply]
that for npm run build 
![four](https://user-images.githubusercontent.com/109394049/234397772-42cdc930-ca89-4eda-a134-c2bac545ebf1.PNG)

**Reply by Ahmedsamir190** [reply]
here before the (compiled client and server successfully) will show and go to localhost:3000 still loading 
when the message shows localhost:3000 show refused I think the server was not created correctly
as you see the terminal waited to run something new
![last](https://user-images.githubusercontent.com/109394049/234402187-87bfe5d2-9e75-46e0-bc6e-d93273a22083.PNG)

**Reply by icyJoseph** [reply]
From the first screenshot, it'd seem that `npm run dev` is crashing out. That's why you can't reach it at localhost:3000 -

The build error is just because of `.babelrc` - which is used to opt-out from SWC. Since that didn't work, you probably can remove the `.babelrc` file.

So far we have made sure you have the right Node.js version for your system, and that SWC is not the issue. 

If you do: `npm list -g --depth 0`, what do you see?

**Reply by Ahmedsamir190** [reply]
![five](https://user-images.githubusercontent.com/109394049/234545672-7ae43115-0eb3-4206-acc9-ebf184ee3341.PNG)

**Reply by icyJoseph** [reply]
Well, this is going nowhere, I thought maybe that Next was installed globally, but that's not the case. Is there any particular reason for you to use the built-in vscode terminal? have you tried with CMD, or PowerShell outside vscode?

**Reply by Ahmedsamir190** [reply]
No reason to use the terminal I indeed I tried to use cmd outside vscode but the problem still
![six](https://user-images.githubusercontent.com/109394049/234551261-ff80e21b-50eb-455b-8f58-20f040888138.PNG)

**Reply by Ahmedsamir190** [reply]
That for removing babelrc file and try npm run dev 
![seven](https://user-images.githubusercontent.com/109394049/234551812-13782cda-8a29-4a9c-8294-ee18ce54d737.PNG)

**Reply by icyJoseph** [reply]
Well, besides trying to set `NODE_OPTIONS='--inspect'`, to try and see if we can figure out what is breaking, https://nextjs.org/docs/advanced-features/debugging#server-side-code - I think I am out of ideas. 

I hope it is not something specific to the OS version.

**Reply by Ahmedsamir190** [reply]
I don't know if I make these steps correctly or not 
![eight](https://user-images.githubusercontent.com/109394049/234560752-3b5d4d4b-b602-4ece-ace3-93af1bbd433d.PNG)
![nine](https://user-images.githubusercontent.com/109394049/234560761-3224711d-d381-41fc-aa23-a4f9415acc32.PNG)

**Reply by Ahmedsamir190** [reply]
I want to thank you to try to help me

**Reply by icyJoseph** [reply]
Interesting, try with `NODE_OPTIONS='--inspect-brk'`, and to see the debugger, you have to click on the Node icon right above your browser console (to the left of `Element`)

**Reply by Ahmedsamir190** [reply]
![ten](https://user-images.githubusercontent.com/109394049/234565441-50836f68-deea-4148-8dd0-c02304fcfa3a.PNG)
